### PR TITLE
document dune files and makefile

### PR DESCRIPTION
**Pull request by jchavarri**
cc @davesnx

### CODE REVIEW

File: src/dune [hunk 1]
```diff
@@ -1,10 +1,28 @@
...
+;; https://dune.readthedocs.io/en/stable/melange.html#melange-emit
 (melange.emit
+
+ ;; The `target` field is used by Dune to put all JavaScript artifacts in a specific folder inside `_build/default`
+ ;; This field `target` allows to have multiple `melange.emit` stanzas in the `dune` file
+ ;; if, for example, we want to produce ES6 and CommonJS output from the same sources.
  (target output)
+
+ ;; The `alias` field can be used later on to build just the artifacts from this `melange.emit`
+ ;; by calling `dune build @react`
```
Comment (davesnx) [inline src/dune:12]: I would add `By default the @melange alias is created`

File: src/dune [hunk 2]
```diff
@@ -1,10 +1,28 @@
...
  (alias react)
+
+ ;; Dune will generate rules to generate JavaScript targets from the libraries sources as well
  (libraries reason-react world)
```
Comment (davesnx) [inline src/dune:16]: I would phrase it differently, like:

"here's the list of dependencies of the stanza". In this case (being a melange.emit), dune will look into those dependencies and generate JavaScript targets from those libraries as well. 

Caveat: those libraries need to be specified with `(modes melange)`

File: src/dune (hunk 1 above)
Comment (davesnx) [reply src/dune:12]: Meh, is on the docs.

**Review by davesnx — APPROVED** [review]

File: src/dune [hunk 3]
```diff
@@ -1,10 +1,28 @@
+;; `melange.emit` is a Dune stanza that will produce build rules to generate
+;; JavaScript files from sources using the Melange compiler
+;; https://dune.readthedocs.io/en/stable/melange.html#melange-emit
 (melange.emit
+
+ ;; The `target` field is used by Dune to put all JavaScript artifacts in a specific folder inside `_build/default`
+ ;; This field `target` allows to have multiple `melange.emit` stanzas in the `dune` file
+ ;; if, for example, we want to produce ES6 and CommonJS output from the same sources.
```
Comment (rolandpeelen) [inline src/dune:8]: Could we include something that will
Make the output js files sit right next to the .re files? I’ve found the ‘promote until clean’ to be a bit weird, sometimes creating a new node_modules folder in my source files and stuff like that - perhaps I’m doing that wrong. An example would be super helpful 🙌

Comment (jchavarri) [reply src/dune]: I am not sure it is possible to produce js files right next to the re files right now. They would be inside a folder including the path to dune file + target in it.

E.g. if the dune file and sources are in `src/foo/`, and target is `bar`, the js files would be placed in `src/foo/bar/src/foo`. There is some more info in https://melange.re/v2.0.0/build-system/#javascript-artifacts-layout.

@anmonteiro [looked into](https://github.com/ocaml/dune/pull/6541) removing `target` but there are some limitations in dune at the moment that prevent it.

Besides that specific thing with paths, one can use promotion normally. I added an commented example in I am not sure it is possible to produce js files right next to the re files right now. They would be inside a folder including the path to dune file + target in it.

E.g. if the dune file and sources are in `src/foo/`, and target is `bar`, the js files would be placed in `src/foo/bar/src/foo`. There is some more info in https://melange.re/v2.0.0/build-system/#javascript-artifacts-layout.

@anmonteiro [looked into](https://github.com/ocaml/dune/pull/6541) removing `target` but there are some limitations in dune at the moment that prevent it, let me know if it helps!

File: src/dune (hunk 2 above)
Comment (jchavarri) [reply src/dune:19]: https://github.com/melange-re/melange-opam-template/pull/26/commits/9ade436272020301a0278344fe06ea9016e4e3c9

File: src/dune (hunk 1 above)
Comment (jchavarri) [reply src/dune]: https://github.com/melange-re/melange-opam-template/pull/26/commits/cf844eeadb76c103f7a49b4b8d35a8455c8dd86e

**Pull request merged**